server/src/main/java/org/apache/kafka/server/config/KafkaConfig.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.server.config;
+
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.config.SecurityConfig;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.config.internals.BrokerSecurityConfigs;
+
+public class KafkaConfig {
```

Comment:
not sure whether we need another class to collect those ssl/sasl-related configs. If we do need it, could we rename it to `KafkaSecurityConfig` to avoid adding other security-unrelated configs in the future. Personally, a fat class like `KafkaConfig.scala` is a anti-pattern.

Reply:
We can, but we already have `SslConfigs`, `SecurityConfig` and `BrokerSecurityConfigs`. KafkaConfig while it is an anti-pattern it has been acting as one place where we can find all KafkaConfig. 

I think better approach would be moving docs and default values to `SslConfigs`, `SecurityConfig` and `BrokerSecurityConfigs` WDYT?

Reply:
We can rename it to `KafkaSecurityConfig` as a common ground between the two approaches. Having central place for all security configs and break out of KafkaConfig anti-pattern

Reply:
> I think better approach would be moving default values to SslConfigs, SaslConfigs ,SecurityConfig and BrokerSecurityConfigs WDYT?

agreed

> KafkaConfig while it is an anti-pattern it has been acting as one place where we can find all KafkaConfig.

got it. I can buy the purpose. BTW, could we rename it to KafkaConfig"s" and make it be a final class since we don't want to have instance of it.

Reply:
I rename it `KafkaSecurityConfigs` and left a comment that it's a central place for Kafka security configs.